packages/aws-cdk/lib/api/aws-auth/sdk.ts

```diff
@@ -555,14 +555,14 @@ export interface IStepFunctionsClient {
  */
 @traceMemberMethods
 export class SDK {
-  private static readonly accountCache = new AccountAccessKeyCache();
```

Comment:
this doesn't need to be a singleton. The class is always reading/writing from a file anywhere and does not cache values. If we ever wanted to, we can make the cache a singleton but keep the class as an instance per SDK.

Reply:
It could also be made a constructor argument. Same point as for the proxy agent.

Reply:
> It could also be made a constructor argument. Same point as for the proxy agent.

True. There is opportunity for refactoring. Right now I want to keep the changes to the minimum to get this in.